content/en/docs/Writing policies/external-data-sources.md

```diff
@@ -0,0 +1,440 @@
+---
+title: External Data Sources
+description: >
+    Use data from ConfigMaps and the Kubernetes API Server 
+weight: 7
+---
+
+The [Variables](/docs/writing-policies/variables/) section discusses how variables help with create smarter and reusable policy definitions and introduced the concept of a rule [`context`](http://localhost:1313/docs/writing-policies/variables/#variables-from-external-data-sources) that stores all variables.
+
+This section provides details on using ConfigMaps and API Calls to reference external data as variables in policies.
+
+{{% alert title="Note" color="info" %}}
+For improved security and performance, Kyverno is designed to not allow connections to systems other than the cluster Kubernetes API server. Use a separate controller or the [sidecar container pattern](https://kubernetes.io/docs/concepts/workloads/pods/#how-pods-manage-multiple-containers) to fetch data from any source and store it in a `ConfigMap` that can be efficiently used in a policy. This design enables separation of concerns and enforcement of security boundaries.
```

Comment:
I get the note, and it's useful, but this might open up a can of worms when users ask, "ooh, how can I do this???".

Reply:
I think you are referring to the "sidecar" references....yeah, lets remove that. We do not want folks inserting sidecars in the Kyverno controller - using a separate controller seems best.